### PR TITLE
Publicipprefixaugust

### PR DESCRIPTION
**Pull request by anilingle-ms**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 1]
```diff
@@ -65,7 +65,7 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.20.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Kindly update the version in the [netcore csproj](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj) as well

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 2]
```diff
@@ -529,6 +533,14 @@
     <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.PublicIpAddressTests\TestPublicIpStandardSku.json">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+<!-- AAI - Something needs to be done here but i dont understand what
+    <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.PublicIpAddressTests\TestPublicIpAddressCRUDPublicIpPrefix.json">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: seems a total of three tests got added, you'll have to add the three recorded files for the tests.

File: src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1 [hunk 3]
```diff
@@ -214,6 +214,8 @@ CmdletsToExport = 'Add-AzureRmApplicationGatewayAuthenticationCertificate',
                'Test-AzureRmPrivateIPAddressAvailability', 
                'Get-AzureRmPublicIpAddress', 'New-AzureRmPublicIpAddress', 
                'Remove-AzureRmPublicIpAddress', 'Set-AzureRmPublicIpAddress', 
+               'Get-AzureRmPublicIpPrefix', 'New-AzureRmPublicIpPrefix',
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1]: you will need to update the [netcore psd1](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/AzureRM.Network.Netcore.psd1) as well

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 4]
```diff
@@ -58,7 +58,8 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: bump this version as well to 20.1.0

Also, update the [netcore csproj](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/Commands.Network.Netcore.csproj) as well

**Comment by praries880** [comment]
@anilingle-ms  The [netcore build failed](https://travis-ci.org/Azure/azure-powershell/builds/413828350#L1319) because of the missing assembly reference updates in the netcore project.

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj (hunk 1 above)
Comment (anilingle-ms) [reply src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Done. Thank you.

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj (hunk 2 above)
Comment (anilingle-ms) [reply src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Ah got it!

Comment (anilingle-ms) [reply src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Updated

**Comment by praries880** [comment]
@anilingle-ms You need to fix the following help issues : 

Target | Description
-- | --
Get-AzureRmPublicIpPrefix.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/get-azurermpublicipprefix
New-AzureRmPublicIpPrefix.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/new-azurermpublicipprefix
Remove-AzureRmPublicIpPrefix.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/remove-azurermpublicipprefix
Set-AzureRmPublicIpPrefix.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/set-azurermpublicipprefix

**Comment by praries880** [comment]
@anilingle-ms Netcore build failed because the network assembly version needs to be bumped up in the netcode test csproj as well : https://travis-ci.org/Azure/azure-powershell/builds/413850153#L1545

**Comment by praries880** [comment]
@MikhailTryakhov  can you review this as well :)

File: src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1 (hunk 3 above)
Comment (anilingle-ms) [reply src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1]: Done

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj (hunk 4 above)
Comment (anilingle-ms) [reply src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: Done

**Comment by praries880** [comment]
design review : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/116

**Comment by praries880** [comment]
@anilingle-ms 
Issues to fix (found by static analysis)


AssemblyFileName | ClassName | Target | Severity | ProblemId | Description | Remediation
-- | -- | -- | -- | -- | -- | --
D:\Public\az-ps-pub4\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzurePublicIpPrefixCommand | Remove-AzureRmPublicIpPrefix | 1 | 8600 | Cmdlet 'Remove-AzureRmPublicIpPrefix' has no defined output type. | Add an OutputType attribute that declares the type of the object(s)   returned by this cmdlet. If this cmdlet returns no output, please set the   output type to 'bool' and make sure to implement the 'PassThru' parameter.
D:\Public\az-ps-pub4\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzurePublicIpPrefixCommand | Remove-AzureRmPublicIpPrefix | 1 | 8510 | Cmdlet 'Remove-AzureRmPublicIpPrefix' has multiple parameter sets, but no   defined default parameter set. | Define a default parameter set in the cmdlet attribute.
D:\Public\az-ps-pub4\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzurePublicIpPrefixCommand | Remove-AzureRmPublicIpPrefix | 1 | 8600 | Cmdlet 'Remove-AzureRmPublicIpPrefix' has no defined output type. | Add an OutputType attribute that declares the type of the object(s)   returned by this cmdlet. If this cmdlet returns no output, please set the   output type to 'bool' and make sure to implement the 'PassThru' parameter.
D:\Public\az-ps-pub4\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzurePublicIpPrefixCommand | Remove-AzureRmPublicIpPrefix | 1 | 8510 | Cmdlet 'Remove-AzureRmPublicIpPrefix' has multiple parameter sets, but no   defined default parameter set. | Define a default parameter set in the cmdlet attribute.

**Comment by praries880** [comment]
@anilingle-ms  Kindly add a change log entry [here](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/ChangeLog.md#current-release).

**Comment by sphibbs** [comment]
Merging with failing CI per @chandrasekarsrinivasan and @anilingle-ms request.

**Pull request merged**